prov/gni/src/gnix_eq.c

```diff
@@ -338,9 +333,16 @@ DIRECT_FN STATIC ssize_t gnix_eq_read(struct fid_eq *eq, uint32_t *event,
 
 	eq_priv = container_of(eq, struct gnix_fid_eq, eq_fid);
 
+	if ((blocking && !eq_priv->wait) ||
+	    (blocking && eq_priv->attr.wait_obj == FI_WAIT_SET))
```

Comment:
According to the man page eq's and cq's and cntr's can all be opened with FI_WAIT_SET wait_object, but specific functions for these can only be used when using a specific wait_object like FI_WAIT_UNSPEC. The man page for fi_eq_sread/from fi_cq_sread/from and fi_cntr_wait all specifically call out that a FI_WAIT_SET object cannot be used when using them, but you can call fi_wait, or the read/from counterpart.